diff -Naur a/Makefile.nt b/Makefile.nt
--- a/Makefile.nt	2003-09-25 09:19:17.000000000 +0100
+++ b/Makefile.nt	2016-07-04 21:06:35.081483000 +0100
@@ -210,8 +210,8 @@
 	mkdir -p $(BINDIR)
 	mkdir -p $(LIBDIR)
 	cd byterun ; $(MAKEREC) install
-	echo "$(STUBLIBDIR)" > $(LIBDIR)/ld.conf
-	echo "$(LIBDIR)" >> $(LIBDIR)/ld.conf
+	echo "$(shell echo $(STUBLIBDIR)|sed -e 's/\//\\\\/g')"| sed -e "s/$$/\r/" > ld.conf
+	echo "$(shell echo $(LIBDIR)| sed -e 's/\//\\\\/g')"| sed -e "s/$$/\r/"  >> ld.conf
 	cp ocamlc $(BINDIR)/ocamlc.exe
 	cp ocaml $(BINDIR)/ocaml.exe
 	cd stdlib ; $(MAKEREC) install
@@ -288,7 +288,7 @@
 
 utils/config.ml: utils/config.mlp config/Makefile
 	@rm -f utils/config.ml
-	sed -e "s|%%LIBDIR%%|$(LIBDIR)|" \
+	sed -e "s|%%LIBDIR%%|$(shell echo $(LIBDIR)|sed -e 's/\//\\\\\\\\\\\\\\\\/g')|" \
             -e "s|%%BYTERUN%%|ocamlrun|" \
             -e 's|%%CCOMPTYPE%%|$(CCOMPTYPE)|' \
             -e "s|%%BYTECC%%|$(BYTECC) $(BYTECCCOMPOPTS)|" \
diff -Naur a/ocamlbuild/ocaml_specific.ml b/ocamlbuild/ocaml_specific.ml
--- a/ocamlbuild/ocaml_specific.ml	2007-05-10 14:32:07.000000000 +0100
+++ b/ocamlbuild/ocaml_specific.ml	2016-07-05 10:14:57.941254600 +0100
@@ -406,7 +406,7 @@
 flag ["ocaml"; "doc"; "docfile"; "extension:texi"] (A"-texi");;
 
 (** Ocamlbuild plugin for it's own building *)
-let install_lib = lazy (try Sys.getenv "INSTALL_LIB" with Not_found -> !*stdlib_dir/"ocamlbuild" (* not My_std.getenv since it's lazy*)) in
+let install_lib = lazy (try Sys.getenv "INSTALL_LIB" with Not_found -> Filename.concat !*stdlib_dir "ocamlbuild" (* not My_std.getenv since it's lazy*)) in
 let install_bin = lazy (My_std.getenv ~default:"/usr/local/bin" "INSTALL_BIN") in
 file_rule "ocamlbuild_where.ml"
   ~prod:"%ocamlbuild_where.ml"
diff -Naur a/patches/gpr#658.patched b/patches/gpr#658.patched
--- a/patches/gpr#658.patched	1970-01-01 01:00:00.000000000 +0100
+++ b/patches/gpr#658.patched	2016-07-05 13:54:02.412191200 +0100
@@ -0,0 +1 @@
+Applied
