diff -Naur a/config/Makefile.msvc b/config/Makefile.msvc
--- a/config/Makefile.msvc	2016-07-08 15:42:56.156715400 +0100
+++ b/config/Makefile.msvc	2016-07-08 15:40:47.702426400 +0100
@@ -97,11 +97,11 @@
 
 ### Build partially-linked object file
 PARTIALLD=lib /nologo
-PACKLD=ld -r --oformat pe-i386
+PACKLD=i686-w64-mingw32-ld -r 
 
 ### nm and objcopy are missing
-BINUTILS_NM=nm
-BINUTILS_OBJCOPY=objcopy
+BINUTILS_NM=i686-w64-mingw32-nm
+BINUTILS_OBJCOPY=i686-w64-mingw32-objcopy
 
 ############# Configuration for the contributed libraries
 
