diff -Naur a/config/Makefile.msvc b/config/Makefile.msvc
--- a/config/Makefile.msvc	2005-02-02 15:39:40.000000000 +0000
+++ b/config/Makefile.msvc	2016-07-04 20:10:58.515076700 +0100
@@ -97,11 +97,11 @@
 
 ### Build partially-linked object file
 PARTIALLD=lib /nologo /debugtype:cv
-PACKLD=ld -r --oformat pe-i386
+PACKLD=i686-w64-mingw32-ld -r 
 
 ### nm and objcopy are missing
-BINUTILS_NM=nm
-BINUTILS_OBJCOPY=objcopy
+BINUTILS_NM=i686-w64-mingw32-nm
+BINUTILS_OBJCOPY=i686-w64-mingw32-objcopy
 
 ############# Configuration for the contributed libraries
 
