diff -Naur a/config/Makefile.msvc64 b/config/Makefile.msvc64
--- a/config/Makefile.msvc64	2007-03-01 09:50:58.000000000 +0000
+++ b/config/Makefile.msvc64	2016-06-19 20:58:56.914192400 +0100
@@ -89,7 +89,8 @@
 DLLCCCOMPOPTS=/Ox /MD -DCAML_DLL
 
 ### Libraries needed
-EXTRALIBS=bufferoverflowu.lib
+#EXTRALIBS=bufferoverflowu.lib  # for the old PSDK compiler only
+EXTRALIBS=
 BYTECCLIBS=advapi32.lib $(EXTRALIBS)
 NATIVECCLIBS=advapi32.lib $(EXTRALIBS)
 
