diff -Naur a/ocamldoc/Makefile.nt b/ocamldoc/Makefile.nt
--- a/ocamldoc/Makefile.nt	2003-08-29 13:15:41.000000000 +0100
+++ b/ocamldoc/Makefile.nt	2016-06-16 21:26:07.917805500 +0100
@@ -195,14 +195,14 @@
 	$(OCAMLOPT) -a -o $@ $(LINKFLAGS) $(OCAMLCMXFILES) $(LIBCMXFILES)
 
 odoc_crc.ml: $(CMIFILES)
-	$(EXTRAC_CRC) $(INCLUDES)\
+	$(EXTRAC_CRC) $(INCLUDES) \
 	Arg Arith_status Array Big_int Buffer Callback Char Digest Dynlink \
 	Filename Format Gc Genlex Hashtbl \
-	Lazy Lexing List Map Marshal Nat Nativeint\
-	Num Obj CamlinternalOO Outcometree Parsing Pervasives Printexc\
-	Printf Profiling Queue Random Ratio\
-	Set Sort Stack Std_exit Str Stream\
-	String Sys Topdirs Toploop Unix Weak\
+	Lazy Lexing List Map Marshal Nat Nativeint \
+	Num Obj CamlinternalOO Outcometree Parsing Pervasives Printexc \
+	Printf Profiling Queue Random Ratio \
+	Set Sort Stack Std_exit Str Stream \
+	String Sys Topdirs Toploop Unix Weak \
 	Printast \
 	Ident \
 	Tbl \
@@ -211,8 +211,8 @@
 	Clflags \
 	Warnings \
 	Ccomp \
-	Linenum\
-	Location\
+	Linenum \
+	Location \
 	Longident \
 	Syntaxerr \
 	Parser \
@@ -258,14 +258,14 @@
 	Main_args \
 	Asttypes \
 	Depend \
-	Odoc_global Odoc_args Odoc_info Odoc_messages Odoc_types\
-	Odoc_misc Odoc_text_parser Odoc_text_lexer\
-	Odoc_text Odoc_comments_global Odoc_parser\
-	Odoc_lexer Odoc_comments Odoc_name Odoc_parameter\
-	Odoc_value Odoc_type Odoc_exception Odoc_class\
-	Odoc_module Odoc_str Odoc_args Odoc_env\
-	Odoc_sig Odoc_ast Odoc_control Odoc_inherit\
-	Odoc_search Odoc_cross Odoc_merge Odoc_analyse\
+	Odoc_global Odoc_args Odoc_info Odoc_messages Odoc_types \
+	Odoc_misc Odoc_text_parser Odoc_text_lexer \
+	Odoc_text Odoc_comments_global Odoc_parser \
+	Odoc_lexer Odoc_comments Odoc_name Odoc_parameter \
+	Odoc_value Odoc_type Odoc_exception Odoc_class \
+	Odoc_module Odoc_str Odoc_args Odoc_env \
+	Odoc_sig Odoc_ast Odoc_control Odoc_inherit \
+	Odoc_search Odoc_cross Odoc_merge Odoc_analyse \
 	Odoc_dag2html Odoc_ocamlhtml Odoc_html Odoc_to_text \
 	Odoc_latex_style Odoc_latex Odoc_man Odoc_texi Odoc_scan > $@
 
